tests/python/unittest/test_optimizer.py

```diff
@@ -399,11 +395,7 @@ def update(self, index, weight, grad, state):
             if self.momentum == 0.0:
                 weight32[:] += -lr * (grad32 + wd * weight32)
             else:
-                mom[:] *= self.momentum
-                weight32[:] -= self.momentum * mom[:]
-                grad32 += wd * weight32
-                grad32 *= lr
-                weight32[:] -= (self.momentum + 1) * grad32
+                weight32[:] += (self.momentum**2 * mom) - lr*(self.momentum+1)*(grad32 + wd*weight32)
```

Comment:
Should multiply `lr` for the first term. `weight32[:] += lr*(self.momentum**2 * mom) - lr*(self.momentum+1)*(grad32 + wd*weight32)`